src/modules/pos/validator/components/ClaimRewardsForm/ClaimRewardsForm.test.js

```diff
@@ -0,0 +1,69 @@
+import { renderWithQueryClient } from 'src/utils/testHelpers';
+import { fireEvent, screen } from '@testing-library/react';
+import mockSavedAccounts from '@tests/fixtures/accounts';
+import { useCurrentAccount } from '@account/hooks';
+import { mockRewardsClaimableWithToken } from '@pos/reward/__fixtures__';
+import useTransactionPriority from '@transaction/hooks/useTransactionPriority';
+import useTransactionFeeCalculation from '@transaction/hooks/useTransactionFeeCalculation';
+import { useRewardsClaimable } from '@pos/reward/hooks/queries';
+import ClaimRewardsForm from './index';
+
+jest.mock('@account/hooks/useDeprecatedAccount', () => ({
+  useDeprecatedAccount: jest.fn().mockReturnValue({
+    isSuccess: true,
+    isLoading: false,
+  }),
+}));
+jest.mock('@transaction/hooks/useTransactionPriority');
+jest.mock('@block/hooks/queries/useLatestBlock');
+jest.mock('@transaction/hooks/useTransactionFeeCalculation');
+jest.mock('@transaction/api');
+
+jest.mock('@account/hooks/useCurrentAccount');
+jest.mock('@pos/reward/hooks/queries');
+
+describe('ClaimRewardsForm', () => {
+  useCurrentAccount.mockReturnValue([mockSavedAccounts[0]]);
+  useRewardsClaimable.mockReturnValue({ data: mockRewardsClaimableWithToken });
+  useTransactionPriority.mockImplementation(() => [
+    { selectedIndex: 1 },
+    () => {},
+    [
+      { title: 'Low', value: 0.001 },
+      { title: 'Medium', value: 0.005 },
+      { title: 'High', value: 0.01 },
+      { title: 'Custom', value: undefined },
+    ],
+  ]);
+  useTransactionFeeCalculation.mockImplementation(() => ({
+    fee: { value: '0.1' },
+    maxAmount: 0.1,
+    minFee: 0.001,
+  }));
+
+  const nextStep = jest.fn();
+
+  const props = {
+    nextStep,
+  };
+
+  it('Should display properly', async () => {
+    renderWithQueryClient(ClaimRewardsForm, props);
+    expect(screen.getByRole('heading', { name: 'Claim rewards' })).toBeTruthy();
+    expect(
+      screen.getByText(
+        'Below are the details of your reward balances, you can continue to claim your rewards and they will be transferred to your wallet balance.'
+      )
+    ).toBeTruthy();
+    mockRewardsClaimableWithToken.data.forEach(({ tokenName }) => {
+      expect(screen.getAllByText(tokenName)[0]).toBeTruthy();
```

Comment:
we could also assert for the rewards amount?

Reply:
Done